lib/pbench/server/api/resources/users_api.py

```diff
@@ -326,36 +370,24 @@ def get(self, username):
                     }
         """
         try:
-            user, verified = self.auth.verify_user(username)
-        except Exception:
-            self.logger.exception("Exception occurred during verifying the user")
-            abort(500, message="INTERNAL ERROR")
-
-        # TODO: Check if the user has the right privileges
-        if verified:
-            response_object = user.get_json()
-            return make_response(jsonify(response_object), 200)
-        elif user.is_admin():
-            try:
-                target_user = User.query(username=username)
-                response_object = target_user.get_json()
-                return make_response(jsonify(response_object), 200)
-            except Exception:
-                self.logger.exception(
-                    "Exception occurred while querying the user. Username: {}", username
-                )
-                abort(500, message="INTERNAL ERROR")
-        else:
-            self.logger.warning(
-                "User {} is not authorized to get user {}.", user.username, username,
+            target_user, http_status, http_message = self.get_valid_target_user(
+                target_username, "GET"
             )
-            abort(
-                403,
-                message=f"Not authorized to access information about user {username}",
+        except Exception:
+            self.logger.exception(
+                "Exception occurred in GET request from user {!r} while validating target user {!r}",
+                self.auth.token_auth.current_user().username,
+                target_username,
             )
+            abort(HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR")
```

Comment:
Do we really need/want this `try`/`except` block here?  I don't think we expect `get_valid_target_user()` to raise or propagate any exceptions; so, if one happens, it's probably OK to just let it blow by to the catch-all handler.

Reply:
(Ditto in PUT and DELETE.)

Reply:
Well there is actually a potential line of code in `get_valid_target_user` that can cause an exception where we query the target_user against a real database so I guess it's probably okay to have this catch block

Reply:
What's the line and potential exception?

Reply:
We query the target user as `target_user = User.query(username=target_username)` if the current user is an admin on line 321

Reply:
Under what circumstances would that result in an exception?  In particular, what's the likelihood of that exception occurring, and what are the costs or what do we lose if we handle it generically using the last-chance handler instead of handling it here?

This code would be simpler and much easier to read without this `try` block.  If the block isn't likely to produce sufficient value, we should remove it, since it does come with a cost.

Reply:
I agree the likelihood is less but it could result in exception if the db instance is down or if we are in the middle of some migrations. By the way are you suggesting the following 
```
result = self.get_valid_target_user(target_username, "GET")
if not result.target_user:
    abort(result.http_status, message=result.http_message)
response_object = result.target_user.get_json()
return make_response(jsonify(response_object), HTTPStatus.OK)
```
or this
```
try:
  result = self.get_valid_target_user(target_username, "GET")
  if not result.target_user:
      abort(result.http_status, message=result.http_message)
  response_object = result.target_user.get_json()
  return make_response(jsonify(response_object), HTTPStatus.OK)
except Exception:
    self.logger.exception(
        "Exception occurred in GET request from user {!r} while validating target user {!r}",
        self.auth.token_auth.current_user().username,
        target_username,
    )
    abort(HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR")
```

Reply:
I'm inclined to suggest the first code blob -- the one with no `try` block.

If there is a database error, then the user will receive an `INTERNAL_SERVER_ERROR` from the catch-all exception handler in either case, so we don't need the `try` block for that.  I'm fairly sure that the log will receive the full traceback, so we'll know exactly what happened.  The only thing that we'll miss out on is the two usernames, which, if it's a problem with accessing the database itself, presumably don't matter.

So, it seems like we're giving up nothing and getting simpler, easier-to-read code in return.

Reply:
Which catch-all exception handler are you pointing to if there is no try catch block?

Reply:
The one in the caller of `UserAPI.get()` -- Flask!  :-D

Reply:
Ah okay